profiling/src/profile/mod.rs

```diff
@@ -986,9 +1036,15 @@ impl TryFrom<&Profile> for pprof::Profile {
             })
             .collect();
 
+        let samples = timestamped_samples?
+            .into_iter()
+            .flatten()
+            .chain(aggregated_samples?)
+            .collect();
```

Comment:
I'm not quite sure about the `?` usage here -- what happens if there are only timestamped samples or only aggregated samples?

Reply:
If either of the `timestamped_samples` or `aggregates_samples` had an error during conversion, the `?` will propagate the error upwards.  If neither had an error, then the iter will take the 0 or more elements from `timestamped_samples`, and then append the 0 or more elements from `aggregates_samples`, and then resolve to a list with both sets of samples in it.

Reply:
Ah, great, that makes a lot of sense! Thanks for putting up with my Rust noobness :)

(I was reading it with my Ruby eyes and suspecting the `?` may be similar to Ruby's `&.` which could have made this a bug, but now that you've explained it the `?` is nothing like the Ruby operator and I was just really confused :D )